File: src/main/js/funk/ioc/injectIn.js
funk.ioc = funk.ioc || {};
funk.ioc.InjectIn = function(value, module){
    if(value === null) {
        throw new funk.error.ArgumentError("Given type must not be null.");
    } else if(module === null){
        throw new funk.error.ArgumentError("Given module type must not be null");
    }
    return funk.ioc.Injector.moduleOf(module).getInstance(value);
};

funk.ioc.injectIn = funk.ioc.InjectIn;

// Alias
var injectIn = funk.ioc.InjectIn;

File: src/main/dart/funk/ioc/injectIn.dart
injectIn(Type type, Type module) {
  if(type == null) {
    throw new ArgumentError("Given type must not be null.");
  } else if(module == null) {
    throw new ArgumentError("Given module type must not be null.");
  }
  
  final Injector injector = new Injector();
  return injector.moduleOf(module).getInstance(type);
}
